tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -1535,6 +1535,71 @@ class StaticAlgorithmSelector : public nvinfer1::IAlgorithmSelector {
                         const nvinfer1::IAlgorithm* const* algoChoices,
                         int32_t nbAlgorithms) override {}  // do nothing
 };
+
+enum class Tactic : int64_t { kINVALID_TACTIC = int64_t(0xD15EA5EDD15EA5ED) };
+
+enum class LayerImpl : int64_t {
+  kSHUFFLE = 16,
+};
+
+static std::unordered_set<int64> trt7_bad_tactics{
+    -5927686925093575778, -3848538574386518527, -959009792490796596};
+
+class FlakyAlgorithmRejector : public nvinfer1::IAlgorithmSelector {
+ public:
+  FlakyAlgorithmRejector() = default;
+  ~FlakyAlgorithmRejector() = default;
```

Comment:
We already have StaticAlgorithmSelector and it would nice to only use one selector. Here is my suggestion on how to achieve that through extending StaticAlgorithmSelector and renaming it to AlgorithmSelector (or TftrtAlgorithmSelector):
(0) we will only use AlgorithmSelector for TRT >=7.
(1) the AlgorithmSelector keep a list of allowed_tactics and disallowed_tactics.
(2) allowed_tactics is initialized through the environment variable TF_TRT_FIXED_ALGORITHM_ID, which currently only allow one tactics. You may want to extend this to allow multiple tactics, but this is not necessary for this PR. 
(3) disallowed_tactics is initialized to the list of known bad TRT7 tactics, if the compiled TRT version match that.
(4) we probably need to do a sanity check to make sure allowed_tactics and disallowed_tactics , if both are not empty, should not contradict each other.
(5) when allowed_tactics is EMPTY, it has a special meaning, that is all tactics that are not in disallowed_tactics are allowed. With this we can express this logic in "selectAlgorithms":
    is allowed_tactics is not EMPTY, a tactics has to be in the list to be selected.
    if a tactics is in diasllowed_tactics, it has to be rejected.

Reply:
Added initial changes to incorporate above suggestions.

Reply:
Updated PR description and Title as per suggestion.